appleseedhq/appleseed
# Deep overhaul of SSS implementation

Please review but **do not merge**.

Suggested review plan:

1. Check the new `renderer::BSSRDF` and `renderer::SeparableBSSRDF` interfaces.
2. Check the generic sampling code in `separablebssrdf.cpp`.
3. Check the new `renderer::BSSRDFSample` class.
4. Check the way subsurface scattering now fits in `pathtracer.h`.
5. Check the changes in `ptlightingengine.cpp`.
6. Review the changes in the individual BSSRDF models.
7. Check the changes made in `test_sss.cpp`.
8. Check the test suite changes.
9. Compare the new reference images to the old ones.
